pkg/pillar/cmd/downloader/download.go

```diff
@@ -67,6 +67,15 @@ func download(ctx *downloaderContext, trType zedUpload.SyncTransportType,
 			asize, osize, progress := resp.Progress()
 			log.Infof("Update progress for %v: %v/%v",
 				resp.GetLocalName(), asize, osize)
+			// sometime, the download goes to an infinite loop,
+			// showing it has downloaded, more than it is supposed to
+			// aborting download, marking it as an error
+			if asize > osize {
+				errStr := fmt.Sprintf("%s, download Error, downloaded %v out of %v",
```

Comment:
Shouldn't the message say "downloaded more than 100% which is impossible. Cancelling".
The current message looks like a non-error until the user compares that asize is greater than osize in the message.

Reply:
Sure, will do.